### PR TITLE
Fix panic in teams API when requesting members

### LINKED ISSUE #19359: Panic when requesting team members API
### Description

I tried to find the commit resposible for this, but I couldn't find it. I tried commit after April 2 and it always panics. Maybe it is database-related or something like this?

```log
[625188bc-3] router: failed    GET /api/v1/teams/4/members for 10.42.0.72:47414, panic in 40.2ms @ org/team.go:347(org.GetTeamMembers), err=runtime error: index out of range [0] with length 0
...common/middleware.go:71:1() [E] [625188bc-4] PANIC: runtime error: index out of range [0] with length 0
        /usr/lib/go-1.17/src/runtime/panic.go:90 (0x43c474)
                goPanicIndex: panic(boundsError{x: int64(x), signed: true, y: y, code: boundsIndex})
        /home/nathan/projects/Go/gitea/gitea/routers/api/v1/org/team.go:392 (0x1f9186e)
                GetTeamMembers: members[i] = convert.ToUser(member, ctx.Doer)
        /home/nathan/projects/Go/gitea/gitea/modules/web/wrap_convert.go:60 (0x1e354d6)
                convertHandler.func5: t(ctx)
        /home/nathan/projects/Go/gitea/gitea/modules/web/wrap.go:41 (0x1e33a89)
                wrapInternal.func1: done, deferrable := handler(resp, req, others...)
        /usr/lib/go-1.17/src/net/http/server.go:2046 (0x8d68ae)
                HandlerFunc.ServeHTTP: f(w, r)
        /home/nathan/go/pkg/mod/github.com/go-chi/chi/v5@v5.0.7/mux.go:442 (0x163b075)
                (*Mux).routeHTTP: h.ServeHTTP(w, r)
        /usr/lib/go-1.17/src/net/http/server.go:2046 (0x8d68ae)
                HandlerFunc.ServeHTTP: f(w, r)
        /home/nathan/projects/Go/gitea/gitea/modules/web/wrap.go:98 (0x1e3460b)
                MiddleAPI.func1.1: next.ServeHTTP(ctx.Resp, ctx.Req)
        /usr/lib/go-1.17/src/net/http/server.go:2046 (0x8d68ae)
                HandlerFunc.ServeHTTP: f(w, r)
        /home/nathan/projects/Go/gitea/gitea/modules/web/wrap.go:98 (0x1e3460b)
                MiddleAPI.func1.1: next.ServeHTTP(ctx.Resp, ctx.Req)
        /usr/lib/go-1.17/src/net/http/server.go:2046 (0x8d68ae)
                HandlerFunc.ServeHTTP: f(w, r)
        /home/nathan/projects/Go/gitea/gitea/modules/context/api.go:272 (0x1a29766)
                APIContexter.func1.1: next.ServeHTTP(ctx.Resp, ctx.Req)
        /usr/lib/go-1.17/src/net/http/server.go:2046 (0x8d68ae)
                HandlerFunc.ServeHTTP: f(w, r)
        /home/nathan/projects/Go/gitea/gitea/routers/api/v1/api.go:1158 (0x1facaf3)
                securityHeaders.func1.1: next.ServeHTTP(resp, req)
        /usr/lib/go-1.17/src/net/http/server.go:2046 (0x8d68ae)
                HandlerFunc.ServeHTTP: f(w, r)
        /home/nathan/go/pkg/mod/github.com/go-chi/chi/v5@v5.0.7/mux.go:71 (0x1638eac)
                (*Mux).ServeHTTP: mx.handler.ServeHTTP(w, r)
        /home/nathan/go/pkg/mod/github.com/go-chi/chi/v5@v5.0.7/mux.go:314 (0x163a85b)
                (*Mux).Mount.func1: handler.ServeHTTP(w, r)
        /usr/lib/go-1.17/src/net/http/server.go:2046 (0x8d68ae)
                HandlerFunc.ServeHTTP: f(w, r)
        /home/nathan/go/pkg/mod/github.com/go-chi/chi/v5@v5.0.7/mux.go:442 (0x163b075)
                (*Mux).routeHTTP: h.ServeHTTP(w, r)
        /usr/lib/go-1.17/src/net/http/server.go:2046 (0x8d68ae)
                HandlerFunc.ServeHTTP: f(w, r)
        /home/nathan/projects/Go/gitea/gitea/routers/common/middleware.go:79 (0x1e9dc22)
                Middlewares.func2.1: next.ServeHTTP(resp, req)
        /usr/lib/go-1.17/src/net/http/server.go:2046 (0x8d68ae)
                HandlerFunc.ServeHTTP: f(w, r)
        /home/nathan/projects/Go/gitea/gitea/modules/web/routing/logger_manager.go:123 (0x1e2f7af)
                (*requestRecordsManager).handler.func1: next.ServeHTTP(w, req)
        /usr/lib/go-1.17/src/net/http/server.go:2046 (0x8d68ae)
                HandlerFunc.ServeHTTP: f(w, r)
        /home/nathan/go/pkg/mod/github.com/go-chi/chi/v5@v5.0.7/middleware/strip.go:30 (0x1e99098)
                StripSlashes.func1: next.ServeHTTP(w, r)
        /usr/lib/go-1.17/src/net/http/server.go:2046 (0x8d68ae)
                HandlerFunc.ServeHTTP: f(w, r)
        /home/nathan/go/pkg/mod/github.com/chi-middleware/proxy@v1.1.1/middleware.go:37 (0x1e95956)
                ForwardedHeaders.func1.1: h.ServeHTTP(w, r)
        /usr/lib/go-1.17/src/net/http/server.go:2046 (0x8d68ae)
                HandlerFunc.ServeHTTP: f(w, r)
        /home/nathan/projects/Go/gitea/gitea/routers/common/middleware.go:32 (0x1e9da72)
                Middlewares.func1.1: next.ServeHTTP(context.NewResponse(resp), req.WithContext(ctx))
        /usr/lib/go-1.17/src/net/http/server.go:2046 (0x8d68ae)
                HandlerFunc.ServeHTTP: f(w, r)
        /home/nathan/go/pkg/mod/github.com/go-chi/chi/v5@v5.0.7/mux.go:88 (0x1638e61)
                (*Mux).ServeHTTP: mx.handler.ServeHTTP(w, r)
        /home/nathan/projects/Go/gitea/gitea/modules/web/route.go:200 (0x1e32ecd)
                (*Route).ServeHTTP: r.R.ServeHTTP(w, req)
        /usr/lib/go-1.17/src/net/http/server.go:2878 (0x8d9e1a)
                serverHandler.ServeHTTP: handler.ServeHTTP(rw, req)
        /usr/lib/go-1.17/src/net/http/server.go:1929 (0x8d5987)
                (*conn).serve: serverHandler{c.server}.ServeHTTP(w, w.req)
        /usr/lib/go-1.17/src/runtime/asm_amd64.s:1581 (0x473b80)
                goexit: BYTE    $0x90   // NOP
```

(Can't reproduce on try.gitea.io because I can't use orgs there)

### Gitea Version

almost latest `main`

### Can you reproduce the bug on the Gitea demo site?

No

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

compiled from source, from command line

### Database

MySQL

### PR DESCRIPTION
**Pull request by delvh**
As title.

Should fix #19359

~~Do we backport API errors?~~
Unnecessary in this case, error does not exist yet on v1.16 branch.
If I had to guess, one of the many `move xxx from models/ to models/xxx/` refactorings were the cause of this bug.

### CODE REVIEW

**Comment by delvh** [comment]
CI is dead, that's the reason why the build fails.

**Review by Gusted — APPROVED** [review]
If this code also exist in the release/1.16 branch it should be backported.

**Review by lunny — APPROVED** [review]

**Review by lunny — CHANGES_REQUESTED** [review]
No, you can't do both. I think you can just change only line 390.

File: routers/api/v1/org/team.go [hunk 1]
```diff
@@ -387,9 +387,9 @@ func GetTeamMembers(ctx *context.APIContext) {
 		return
 	}
 
-	members := make([]*api.User, len(ctx.Org.Team.Members))
-	for i, member := range teamMembers {
-		members[i] = convert.ToUser(member, ctx.Doer)
+	members := make([]*api.User, len(teamMembers))
+	for _, member := range teamMembers {
```
Comment (lunny) [inline routers/api/v1/org/team.go:391]:
```suggestion
	for i, member := range teamMembers {
		members[i] = convert.ToUser(member, ctx.Doer)
```

File: routers/api/v1/org/team.go [hunk 2]
```diff
@@ -387,9 +387,9 @@ func GetTeamMembers(ctx *context.APIContext) {
...
+		members = append(members, convert.ToUser(member, ctx.Doer))
```
Comment (lunny) [inline routers/api/v1/org/team.go]:
```suggestion
	for i, member := range teamMembers {
		members[i] = convert.ToUser(member, ctx.Doer)
```

**Comment by delvh** [comment]
I mean, what's the benefit?
Both should work, with the only difference that with `append` another error like this cannot happen...
But I can apply it of course.

**Review by lunny — APPROVED** [review]

**Comment by lunny** [comment]
> I mean, what's the benefit? Both should work, with the only difference that with `append` another error like this cannot happen... But I can apply it of course.

Both should work, but you should only chose one. You can't chose both, otherwise the slice will have 2 times elements and the first part are empty elements.

**Review by wxiaoguang — APPROVED** [review]

**Pull request merged**